bundles/persistence/org.openhab.persistence.jdbc/java/org/openhab/persistence/jdbc/internal/JdbcConfiguration.java

```diff
@@ -0,0 +1,321 @@
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.persistence.jdbc.internal;
+
+import java.util.Enumeration;
+import java.util.Map;
+import java.util.Properties;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.apache.commons.collections.iterators.IteratorEnumeration;
+import org.apache.commons.lang.StringUtils;
+import org.openhab.persistence.jdbc.db.JdbcBaseDAO;
+import org.openhab.persistence.jdbc.utils.MovingAverage;
+import org.openhab.persistence.jdbc.utils.StringUtilsExt;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * 
+ * @author Helmut Lehmeyer
+ * @since 1.8.0
+ */
+public class JdbcConfiguration {
+	private static final Logger logger = LoggerFactory.getLogger(JdbcConfiguration.class);
+
+	private static final Pattern EXTRACT_CONFIG_PATTERN = Pattern.compile("^(.*?)\\.([0-9.a-zA-Z]+)$");
+	private static final String DB_DAO_PACKAGE = "org.openhab.persistence.jdbc.db.Jdbc";
+
+	private JdbcBaseDAO dBDAO = null;
+	private String dbName = null;
+	boolean dbConnected = false;
+	boolean driverAvailable = false;
+
+	private String serviceName;
+	private String name = "jdbc";
+
+	//private String url;
+	//private String user;
+	//private String password;
+	private int numberDecimalcount = 3;
+	private boolean tableUseRealItemNames = false;
+	private String tableNamePrefix = "item";
+	private int tableIdDigitCount = 4;
+	private boolean rebuildTableNames = false;
+
+	private int errReconnectThreshold = 0;
+
+	public int timerCount = 0;
+	public int time1000Statements = 0;
+	public long timer1000 = 0;
+	public MovingAverage timeAverage50arr = new MovingAverage(50);
+	public MovingAverage timeAverage100arr = new MovingAverage(100);
+	public MovingAverage timeAverage200arr = new MovingAverage(200);
+	public boolean enableLogTime = false;
+
+	public JdbcConfiguration(Map<Object, Object> configuration) {
+		logger.debug("JDBC::JdbcConfiguration");
+		updateConfig(configuration);
+	}
+
+	/**
+	 * @{inheritDoc
+	 */
+	public void updateConfig(Map<Object, Object> configuration) {
+		logger.debug("JDBC::updateConfig: configuration.size = " + configuration.size());
+
+		// Database-Url jdbc:h2:./testH2
+		String url = (String) configuration.get("url");
+		Properties parsedURL = StringUtilsExt.parseJdbcURL(url);
+		logger.debug("JDBC::updateConfig: url={}", url);
+		if (StringUtils.isBlank(url) || parsedURL.getProperty("parseValid") == "false") {
+			logger.error("JDBC::updateConfig: url The SQL database URL is missing - please configure the jdbc:url parameter in openhab.cfg");
+		}else{
+			dbName = parsedURL.getProperty("databaseName");
+		}
+
+		// Which DB-Type to use
+		serviceName = parsedURL.getProperty("dbShortcut"); // derby, h2, hsqldb, mariadb, mysql, postgresql, sqlite
+		logger.debug("JDBC::updateConfig: found serviceName = '{}'", serviceName);
+		if (StringUtils.isBlank(serviceName)) {
```

Comment:
minor nitpick: make sure that serviceName is at least two characters, else you will run into IndexOutOfBoundsException a few lines later.